### PR TITLE
AK-28216 Deprecate fortinet credential resources

### PR DESCRIPTION
**Pull request by Alshadex**
Moved fortinet credentials and fortinet instance credentials into fortinet service. Added deprecation messages. Fixed license_type duplication, only required in service_fortinet.

### CODE REVIEW

File: alkira/resource_alkira_credential_fortinet.go [hunk 1]
```diff
@@ -17,6 +17,9 @@ func resourceAlkiraCredentialFortinet() *schema.Resource {
 		Importer: &schema.ResourceImporter{
 			State: schema.ImportStatePassthrough,
 		},
+		DeprecationMessage: "alkira_credential_fortinet has been deprecated. " +
+			"Please specify username and password directly in resource service_fortinet. " +
```
Comment (spikewang) [inline alkira/resource_alkira_credential_fortinet.go:21]: the formal resource should be `alkira_service_fortinet`

File: alkira/resource_alkira_credential_fortinet_instance.go [hunk 2]
```diff
@@ -27,6 +24,9 @@ func resourceAlkiraCredentialFortinetInstance() *schema.Resource {
 			"the Alkira provider will treat the `license_key` field with precedence. \n\n\n " +
 			"You may also use terraform's built in `file` helper function as a literal input for " +
 			"`license_key`. Ex: `license_key = file('/path/to/license/file')`.",
+		DeprecationMessage: "alkira_credential_fortinet_instance has been deprecated. " +
+			"Please specify license_key or license_key_file_path directly in resource service_fortinet. " +
```
Comment (spikewang) [inline alkira/resource_alkira_credential_fortinet_instance.go:28]: Putting ` around license_key and license_key_file.

File: alkira/resource_alkira_service_fortinet.go [hunk 3]
```diff
@@ -68,6 +86,7 @@ func resourceAlkiraServiceFortinet() *schema.Resource {
 								"Required only when licenseType is BRING_YOUR_OWN.",
```
Comment (spikewang) [inline alkira/resource_alkira_service_fortinet.go:86]: This comment needs to be reformed.

Required only when `license_type` is `BRING_YOUR_OWN`.

File: alkira/resource_alkira_service_fortinet_helper.go [hunk 4]
```diff
@@ -1,18 +1,26 @@
...
+	"fmt"
 	"log"
+	"os"
 
 	"github.com/alkiranet/alkira-client-go/alkira"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-func expandFortinetInstances(in []interface{}) []alkira.FortinetInstance {
+func expandFortinetInstances(licenseType string, in []interface{}, m interface{}) ([]alkira.FortinetInstance, error) {
```
Comment (spikewang) [inline alkira/resource_alkira_service_fortinet_helper.go:13]: hmmm, why licenseType is a special one passed in? This should be part of the config... nope?

**Review by spikewang — CHANGES_REQUESTED** [review]

File: alkira/resource_alkira_service_fortinet_helper.go (hunk 4 above)
Comment (Alshadex) [reply alkira/resource_alkira_service_fortinet_helper.go:13]: LicenseType is required by the API to be passed in when creating the fortinet instance credentials and fortinet service.

**Review by spikewang — APPROVED** [review]

**Pull request merged**